daft/dataframe/dataframe.py

```diff
@@ -158,6 +158,28 @@ def _result(self) -> Optional[PartitionSet]:
         else:
             return self._result_cache.value
 
+    def explain_broadcast(self):
+        """Broadcast the mermaid-formatted plan on the given port (assuming metrics-broadcasting is enabled)."""
+        import socket
+
+        from daft.dataframe.display import MermaidFormatter
+
+        ctx = get_context()
+        with ctx._lock:
```

Comment:
No need to grab lock since we're just reading I think

Reply:
Updated.